iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -1467,23 +1493,6 @@ void LinalgBufferizePass::runOnFunction() {
   if (walkResult.wasInterrupted()) {
     return signalPassFailure();
   }
-
-  // Clean-up scf.for by removing cyclic loop dependainces.
-  // Forward init arguments from outer scf.for loop to the inner loops.
-  funcOp.walk<WalkOrder::PreOrder>([&](scf::ForOp scfForOp) {
-    if (scfForOp.results().empty()) return;
-    if (!llvm::all_of(scfForOp.results(), [](Value result) -> bool {
-          auto resultType = result.getType();
-          if (resultType && resultType.isa<TensorType>()) return true;
```

Comment:
@MaheshRavishankar This was the part that didn't delete the operands when doing `powOp`. not sure why this is failing `powOp` now!

Reply:
This part is not needed at all. Actually the logic for `scf.for` conversion is now more controlled. It only looks for the `RankedTensorType` operands and deals with those. So we dont need to do anything special for mixed operands (it is already assuming mixed operands). Just had a couple of bugs. Fixed now.

Reply:
Awesome!